packages/node-terra/src/utils/terra-helper.ts

```diff
@@ -30,29 +35,70 @@ export function filterEvents(
     filterOrFilters instanceof Array ? filterOrFilters : [filterOrFilters];
   const filteredEvents = [];
   events.forEach((event) => {
-    const fe = {};
-    filters.forEach((filter) => {
-      if (filter.type in event) {
-        fe[filter.type] = event[filter.type];
+    for (const filter of filters) {
+      const msg = event.txInfo.tx.body.messages[event.log.msg_index].toData();
+      if (
+        filter.contract &&
+        (msg['@type'] !== '/terra.wasm.v1beta1.MsgExecuteContract' ||
+          msg.contract !== filter.contract)
+      ) {
+        continue;
+      }
+      if (!(filter.type in event.event)) {
+        continue;
       }
-    });
-    if (Object.keys(fe).length !== 0) {
-      filteredEvents.push(fe);
+      filteredEvents.push(event);
+      break;
     }
   });
   console.log(filteredEvents);
   return filteredEvents;
 }
 
-async function getBlockByHeight(api: TerraClient, height: number) {
-  return api.blockInfo(height).catch((e) => {
+export function filterCalls(
+  calls: TerraCall[],
+  filterOrFilters: SubqlTerraCallFilter | SubqlTerraCallFilter[] | undefined,
+): TerraCall[] {
+  if (
+    !filterOrFilters ||
+    (filterOrFilters instanceof Array && filterOrFilters.length === 0)
+  ) {
+    return calls;
+  }
+
+  const filters =
+    filterOrFilters instanceof Array ? filterOrFilters : [filterOrFilters];
+
+  const filteredCalls = [];
+  for (const call of calls) {
+    const callsToPush = [];
+    for (const filter of filters) {
+      if (filter?.from && filter.from !== call.data.sender) {
+        continue;
+      }
+      if (filter?.contract && filter.contract !== call.data.contract) {
+        continue;
+      }
+      if (!(filter?.function in call.data.execute_msg)) {
+        continue;
+      }
+      callsToPush.push(call);
+    }
+    filteredCalls.push(...callsToPush);
+  }
+  console.log(filteredCalls);
+  return filteredCalls;
+}
+
+async function getBlockByHeight(api: LCDClient, height: number) {
+  return api.tendermint.blockInfo(height).catch((e) => {
     logger.error(`failed to fetch Block ${height}`);
     throw e;
   });
 }
 
 export async function fetchTerraBlocksArray(
-  api: TerraClient,
+  api: LCDClient,
```

Comment:
Like in my previous comment, all these need to be changed back to `TerraClient`. There are various occurrences and `LCDClient` shouldn't be called directly